### ISSUE TITLE
AssertionError: No inf checks were recorded prior to update.

### ISSUE THREAD

**Issue by ioslide** [huggingface/accelerate#357]
when I changed "scaler = GradScaler(enabled=True)" to "scaler = accelerator.scaler"

#scaler.step(optimizer)
#scaler.update()

AssertionError: No inf checks were recorded prior to update.

**Comment by ioslide** [comment]
can u provide a completely examples about "accelerator.scaler"

**Comment by sgugger** [comment]
No, as you should not use it. Mixed precision is entirely done behind the scenes by `Accelerator`, there is no situation where you need to call the `scaler`, especially for to step.
